mifospay/src/main/java/org/mifos/mobilewallet/mifospay/home/ui/ProfileFragment.java

```diff
@@ -133,8 +137,16 @@ public void onUserImageEditClicked() {
 
     @OnClick(R.id.btn_profile_bottom_sheet_action)
     public void onEditProfileClicked() {
+        verifyPasscode();
+    }
+
+    private void verifyPasscode() {
         if (getActivity() != null) {
-            getActivity().startActivity(new Intent(getActivity(), EditProfileActivity.class));
+            Intent intent = new Intent(getActivity(), PassCodeActivity.class);
+            intent.putExtra(PassCodeConstants.PASSCODE_INITIAL_LOGIN, true);
+            intent.putExtra("editProfile", "editProfile");
```

Comment:
Everything in your logic and code seems Ok. But please keep in mind `Code Scalability` and `Proper Attributes Naming` play a very important role to write better quality code. Instead of these hardcoded strings,I really recommend you to first, add the **Extra Key** as` public static final String PASSCODE_NEXT_ACTIVITY = "passcode_next_code_activity";` in `Constants.java` file and for **value**  use `EditProfileActivity.class.getSimpleName()`

Reply:
> Everything in your logic and code seems Ok. But please keep in mind `Code Scalability` and `Proper Attributes Naming` play a very important role to write better quality code. Instead of these hardcoded strings,I really recommend you to first, add the **Extra Key** as` public static final String PASSCODE_NEXT_ACTIVITY = "passcode_next_code_activity";` in `Constants.java` file and for **value** use `EditProfileActivity.class.getSimpleName()`

Sure I will take care of these instructions Thank you :)